plugins/woocommerce/src/Internal/Orders/SourceAttributionController.php

```diff
@@ -0,0 +1,473 @@
+	}
+
+	/**
+	 * Display the origin column in the orders table.
+	 *
+	 * @since x.x.x
+	 *
+	 * @param int $order_id The order ID.
+	 *
+	 * @return void
+	 */
+	private function display_origin_column( $order_id ): void {
+		try {
+			// Ensure we've got a valid order.
+			$order = $this->get_hpos_order_object( $order_id );
+			$this->output_origin_column( $order );
+		} catch ( Exception $e ) {
+			return;
+		}
+	}
+
+	/**
+	 * Output the data for the Origin column in the orders table.
+	 *
+	 * @param WC_Order $order The order object.
+	 *
+	 * @return void
+	 */
+	private function output_origin_column( WC_Order $order ) {
+		$source_type = $order->get_meta( $this->get_meta_prefixed_field( 'type' ) );
+		$source      = $order->get_meta( $this->get_meta_prefixed_field( 'utm_source' ) ) ?: __( '(none)', 'woocommerce' );
+		echo esc_html( $this->get_origin_label( $source_type, $source ) );
+	}
+
+	/**
+	 * Add attribution hidden input fields for checkout & customer register froms.
+	 */
+	private function source_form_fields() {
+		foreach ( $this->fields as $field ) {
+			printf( '<input type="hidden" name="%s" value="" />', esc_attr( $this->get_prefixed_field( $field ) ) );
+		}
+	}
+
+	/**
+	 * Save source data for a Customer object.
+	 *
+	 * @param WC_Customer $customer The customer object.
+	 *
+	 * @return void
+	 */
+	private function set_customer_source_data( WC_Customer $customer ) {
+		// phpcs:ignore WordPress.Security.NonceVerification
+		foreach ( $this->get_source_values( $_POST ) as $key => $value ) {
+			$customer->add_meta_data( $this->get_meta_prefixed_field( $key ), $value );
+		}
+
+		$customer->save_meta_data();
+	}
+
+	/**
+	 * Save source data for an Order object.
+	 *
+	 * @param array    $source_data The source data.
+	 * @param WC_Order $order       The order object.
+	 *
+	 * @return void
+	 */
+	private function set_order_source_data( array $source_data, WC_Order $order ) {
+		foreach ( $source_data as $key => $value ) {
+			$order->add_meta_data( $this->get_meta_prefixed_field( $key ), $value );
+		}
+
+		$order->save_meta_data();
+	}
+
+	/**
+	 * Log a message as a debug log entry.
+	 *
+	 * @param string $message The message to log.
+	 * @param string $method  The method that is logging the message.
+	 * @param string $level   The log level.
+	 */
+	private function log( string $message, string $method, string $level = WC_Log_Levels::DEBUG ) {
+		/**
+		 * Filter to enable debug mode.
+		 *
+		 * @since x.x.x
+		 *
+		 * @param string $enabled 'yes' to enable debug mode, 'no' to disable.
+		 */
+		if ( 'yes' !== apply_filters( 'wc_order_source_attribution_debug_mode_enabled', 'no' ) ) {
+			return;
+		}
+
+		$this->logger->log(
+			$level,
+			sprintf( '%s %s', $method, $message ),
+			array( 'source' => 'woocommerce-order-source-attribution' )
+		);
+	}
+
+	/**
+	 * Check to see if HPOS is enabled.
+	 *
+	 * @return bool
+	 */
+	private function is_hpos_enabled(): bool {
```

Comment:
Would it be an option to use `CustomOrdersTableController::custom_orders_table_usage_is_enabled()` here?

Reply:
✔️ I went with `OrderUtil::custom_orders_table_usage_is_enabled()`